dev-java/batik/batik-1.18.ebuild

```diff
@@ -0,0 +1,247 @@
+			JAVA_SRC_DIR="${module}/src/main/java"
+			java-pkg-simple_src_compile
+		else
+			# This case applies to batik-rasterizer.
+			if [[ -d "${module}/src/main/resources/org" ]]; then
+				jar -cfe "${module}.jar" "${JAVA_MAIN_CLASS}" -C "${module}/src/main/resources" . || die
+			# Else for batik-squiggle (also batik-rasterizer-ext, batik-squiggle-ext)
+			else
+				# Create the JAR file (not possible without adding at least one file).
+				jar -cfe "${module}.jar" "${JAVA_MAIN_CLASS}" -C . README || die
+				zip -d "${module}.jar" "README" || die
+			fi
+		fi
+
+		JAVA_GENTOO_CLASSPATH_EXTRA+=":${module}.jar"
+
+		rm -fr target || die
+	done
+
+	if use doc; then
+		JAVA_SRC_DIR=""
+		JAVA_JAR_FILENAME="ignoreme.jar"
+
+		for module in "${BATIK_MODULES[@]}"; do
+			# Some modules don't have source code
+			if [[ -d "${module}/src/main/java/org" ]]; then
+				JAVA_SRC_DIR+=( "${module}/src/main/java" )
+			fi
+		done
+
+		java-pkg-simple_src_compile
+	fi
+}
+
+src_test() {
+	JAVA_TEST_GENTOO_CLASSPATH="junit-4"
+
+	for module in "${BATIK_MODULES[@]}"; do
+		einfo "Testing ${module}"
+
+		JAVA_TEST_SRC_DIR=""
+		JAVA_TEST_RESOURCE_DIRS=""
+		JAVA_TEST_RUN_ONLY=""
+
+		if [[ -d "${module}/src/test/resources" ]]; then
+			JAVA_TEST_RESOURCE_DIRS="${module}/src/test/resources"
+		fi
+
+		# https://github.com/apache/xmlgraphics-batik/blob/refs/tags/batik-1_14/batik-util/pom.xml#L74-L75
+		if [[ "${module}" == batik-util ]]; then
+			JAVA_TEST_RUN_ONLY="org.apache.batik.util.Base64TestCases"
+		fi
+
+		if [[ -d "${module}/src/test/java/org" ]]; then
+			JAVA_TEST_SRC_DIR="${module}/src/test/java"
+			java-pkg-simple_src_test
+		fi
+	done
+}
+
+src_install() {
+	einstalldocs
+	domenu "batik-squiggle-${SLOT}.desktop"
+	newicon -s scalable \
+		batik-svgbrowser/src/main/resources/org/apache/batik/apps/svgbrowser/resources/init.svg \
+		squiggle-${SLOT}.svg
+
+	for module in "${BATIK_MODULES[@]}"; do
+		JAVA_MAIN_CLASS=$( sed -n 's:.*<mainClass>\(.*\)</mainClass>:\1:p' "${module}/pom.xml" )
+
+		java-pkg_dojar "${module}.jar"
+
+		# Add a launcher if the module has a MainClass.
+		if [[ -n "${JAVA_MAIN_CLASS}" ]]; then
+			java-pkg_dolauncher "${module}-${SLOT}" --main "${JAVA_MAIN_CLASS}"
+		fi
+
+		# Some modules don't have source code
+		if [[ -d "${module}/src/main/java/org" ]]; then
+			if use source; then
+				java-pkg_dosrc "${module}/src/main/java/*"
+			fi
+		fi
+	done
+
+	local java_policy_file="${JAVA_PKG_SHAREPATH}/etc/${PN}.policy"
+	insinto "$(dirname "${java_policy_file}")"
+	newins - "$(basename "${java_policy_file}")" <<- _EOF_
+		grant codeBase "file:${EPREFIX}${JAVA_PKG_JARDEST}/-" {
+			permission java.security.AllPermission;
+		};
+	_EOF_
+	java-pkg_register-environment-variable \
+		gjl_java_args \
+		"\$gjl_java_args -Djava.security.policy=file:${EPREFIX}${java_policy_file}"
+
+	if use doc; then
+		java-pkg_dojavadoc target/api
+	fi
+}
+
+pkg_postinst() {
+	xdg_desktop_database_update
+	xdg_icon_cache_update
```

Comment:
aren't those ran by default?

Reply:
> aren't those ran by default?

apparently not. dropping those lines would lead to:
```
Updating .desktop files database ...
debug: gtk-update-icon-cache is not found
```